voting/tests/common/scenarios.rs

```diff
@@ -317,3 +330,77 @@ pub fn test_vote(ctx: &RwLock<Context>) {
+    for stateful in ctx.write().statefuls.values_mut() {
+        stateful.set_storage(ServiceRef::create_export(Box::new(MockDb::default()) as Box<dyn SubStorageAccess>))
+    }
+
+    let admin: Ed25519KeyPair = Random.generate().unwrap();
+    let admin_key = admin.public();
+
+    ctx.write()
+        .init_genesises
+        .get_mut("general_meeting")
+        .unwrap()
+        .init_genesis(&serde_cbor::to_vec(&admin_key).unwrap());
+
+    let create_meeting = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 12, 18);
+
+    let block_hash: BlockHash = BlockHash::default();
+    let author: Public = Public::default();
+    let validators: Vec<Public> = Vec::new();
+    let extera_data: Bytes = Bytes::default();
+
+    let header = Header::new(block_hash, 1, 0, author, validators.clone(), extera_data.clone());
+    ctx.write().tx_owners.get_mut("general_meeting").unwrap().block_opened(&header).unwrap();
+
+    let creat_meeting_outcome =
+        ctx.write().tx_owners.get_mut("general_meeting").unwrap().execute_transaction(&create_meeting).unwrap();
+    let meeting_id: H256 = serde_cbor::from_slice(&creat_meeting_outcome.events.get(0).unwrap().value).unwrap();
+
+    let voter1: Ed25519KeyPair = Random.generate().unwrap();
+    let header = Header::new(block_hash, 1, 0, author, validators.clone(), extera_data.clone());
+    ctx.write().tx_owners.get_mut("voting").unwrap().block_opened(&header).unwrap();
+
+    let create_vote_paper = tx_vote_paper(
+        voter1.public(),
+        voter1.private(),
+        meeting_id,
+        2,
+        10,
+        String::from("CodeChain"),
+        *voter1.public(),
+    );
+    let vote_paper_execution =
+        ctx.write().tx_owners.get_mut("voting").unwrap().execute_transaction(&create_vote_paper).unwrap();
+
+    let vote_paper_id: H256 = serde_cbor::from_slice(&vote_paper_execution.events.get(0).unwrap().value).unwrap();
+    let signature = ckey::sign("vote".as_ref(), voter1.private());
+    let choice = VoteChoice::Favor;
+    let vote = tx_vote(vote_paper_id, choice.clone(), signature);
+
+    let vote_transaction = ctx.write().tx_owners.get_mut("voting").unwrap().execute_transaction(&vote);
+
+    let general_meeting_id: H256 = serde_cbor::from_slice(&creat_meeting_outcome.events.get(0).unwrap().value).unwrap();
+
+    // This transaction will return an error due to an early request.
+    let publish_result = tx_publish_result(general_meeting_id);
+    let early_publish_transaction = ctx.write().tx_owners.get_mut("general_meeting").unwrap().execute_transaction(&publish_result);
+    assert!(early_publish_transaction.is_err());
+
+    let third_block_hash: BlockHash = BlockHash::default();
+    let third_author: Public = Public::default();
+    let third_set_validators: Vec<Public> = Vec::new();
+    let third_extera_data: Bytes = Bytes::default();
+
+    let header = Header::new(third_block_hash, 29 , 0, third_author, third_set_validators, third_extera_data.clone());
```

Comment:
IMO, no need to use new variables since they are using the default value.
```suggestion
    let header = Header::new(block_hash, 29, 0, author, validators.clone(), extera_data.clone());
```